### PR TITLE
[feature](Cloud) Support create storage vault stmt

### PR DESCRIPTION
**Pull request by ByteYue**
## Proposed changes

Issue Number: close #xxx

This pr is part of `HDFS support for compute storage separation`. This pr introduces the create logic for storage vault. And make one new property name when creating table. Also it would call the rpc interface of Meta Service to check if the corresponding storage vault exists or not along with the add storage vault interface of Meta service. 

You can create one storage vault using the following stmt

```sql
CREATE STORAGE VAULT hdfs_vault PROPERTIES (
   "type"="hdfs",
   "username"="user",
   "password"="passwd",
   "dfs.nameservices" = "my_ha",
   "dfs.ha.namenodes.my_ha" = "my_namenode1, my_namenode2",
   "dfs.namenode.rpc-address.my_ha.my_namenode1" = "nn1_host:rpc_port",
   "dfs.namenode.rpc-address.my_ha.my_namenode2" = "nn2_host:rpc_port",
   "dfs.client.failover.proxy.provider" = "org.apache.hadoop.hdfs.server.namenode.ha.ConfiguredFailoverProxyProvider"
);

```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run feut

File: fe/fe-core/src/main/java/org/apache/doris/catalog/HdfsStorageVault.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    @SerializedName(value = "properties")
+    private Map<String, String> properties;
+
+    public HdfsStorageVault(String name, boolean ifNotExists) {
+        super(name, StorageVault.StorageVaultType.HDFS, ifNotExists);
+        properties = Maps.newHashMap();
+    }
+
+    @Override
+    public void alterMetaService() throws DdlException {
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/HdfsStorageVault.java:65]: move it to storage vault manager

File: fe/fe-core/src/main/java/org/apache/doris/catalog/HdfsStorageVault.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    }
+
+    @Override
+    public Map<String, String> getCopiedProperties() {
+        return Maps.newHashMap(properties);
+    }
+
+    @Override
+    protected void getProcNodeData(BaseProcResult result) {
+        // String lowerCaseType = type.name().toLowerCase();
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/HdfsStorageVault.java:116]: remove if not needed

File: gensrc/proto/cloud.proto [hunk 3]
```diff
@@ -691,20 +713,29 @@ message GetObjStoreInfoRequest {
     optional string cloud_unique_id = 1; // For auth
 };
 
+message AlterHdfsParams {
+        optional string vault_name = 1;
+        optional HdfsParams hdfs = 2;
+    }
```
Comment (gavinchou) [inline gensrc/proto/cloud.proto:719]: bad indention

File: gensrc/proto/cloud.proto [hunk 4]
```diff
@@ -187,6 +190,25 @@ message ObjectStoreInfoPB {
     optional bool sse_enabled = 15;
 }
 
+message StorageVaultPB {
+    optional string id = 1;
+    optional string name = 2;
+    optional HdfsParams hdfs_infos = 3; // HdfsResource
+}
+
+message HdfsParams {
```
Comment (gavinchou) [inline gensrc/proto/cloud.proto:199]: naming  HdfsParams -> HdfsInfo

**Review by gavinchou — CHANGES_REQUESTED** [review]

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

File: gensrc/proto/cloud.proto [hunk 5]
```diff
@@ -187,6 +190,25 @@ message ObjectStoreInfoPB {
     optional bool sse_enabled = 15;
 }
 
+message StorageVaultPB {
+    optional string id = 1;
+    optional string name = 2;
+    optional HdfsInfo hdfs_infos = 3; // HdfsResource
```
Comment (wangshuo128) [inline gensrc/proto/cloud.proto:196]:
```suggestion
    optional HdfsInfo hdfs_info = 3; // HdfsResource
```

File: gensrc/proto/cloud.proto [hunk 6]
```diff
@@ -187,6 +190,25 @@ message ObjectStoreInfoPB {
...
+}
+
+message HdfsInfo {
+    optional string fs_name = 1;
```
Comment (wangshuo128) [inline gensrc/proto/cloud.proto:200]: We should have a field `path` in `HdfsInfo`, rather then in `hdfs_confs`.
`path` is the place to save data, we should use the `path` value to create dir or list files. This is one of the arguments of HDFS client methods.
`hdfs_confs` is the configs we create a HDFS client.

**Review by platoneko — DISMISSED** [review]

**Comment by ByteYue** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**